delta-lake/src/main/java/org/apache/iceberg/BaseMigrateDeltaLakeTableAction.java

```diff
@@ -0,0 +1,322 @@
+      Catalog icebergCatalog,
+      String deltaTableLocation,
+      TableIdentifier newTableIdentifier,
+      Configuration hadoopConfiguration) {
+    this.icebergCatalog = icebergCatalog;
+    this.deltaTableLocation = deltaTableLocation;
+    this.newTableIdentifier = newTableIdentifier;
+    this.hadoopConfiguration = hadoopConfiguration;
+    // TODO: check whether we can retrieve hadoopConfiguration directly
+    this.deltaLog = DeltaLog.forTable(this.hadoopConfiguration, this.deltaTableLocation);
+  }
+
+  @Override
+  public MigrateDeltaLakeTable tableProperties(Map<String, String> properties) {
+    additionalProperties.putAll(properties);
+    return this;
+  }
+
+  @Override
+  public Result execute() {
+    io.delta.standalone.Snapshot updatedSnapshot = deltaLog.update();
+    Schema schema = convertDeltaLakeSchema(updatedSnapshot.getMetadata().getSchema());
+    PartitionSpec partitionSpec = getPartitionSpecFromDeltaSnapshot(schema);
+    // TODO: check whether we need more info when initializing the table
+    Table icebergTable =
+        this.icebergCatalog.createTable(
+            newTableIdentifier,
+            schema,
+            partitionSpec,
+            destTableProperties(
+                updatedSnapshot, this.deltaTableLocation, this.additionalProperties));
+
+    copyFromDeltaLakeToIceberg(icebergTable, partitionSpec);
+
+    Snapshot snapshot = icebergTable.currentSnapshot();
+    long totalDataFiles =
+        Long.parseLong(snapshot.summary().get(SnapshotSummary.TOTAL_DATA_FILES_PROP));
+    LOG.info(
+        "Successfully loaded Iceberg metadata for {} files to {}",
+        totalDataFiles,
+        deltaTableLocation);
+    return new BaseMigrateDeltaLakeTableActionResult(totalDataFiles);
+  }
+
+  /** TODO: check the correctness for nested schema */
+  private Schema convertDeltaLakeSchema(io.delta.standalone.types.StructType deltaSchema) {
+    Type converted =
+        DeltaLakeDataTypeVisitor.visit(deltaSchema, new DeltaLakeTypeToType(deltaSchema));
+    return new Schema(converted.asNestedType().asStructType().fields());
+  }
+
+  private PartitionSpec getPartitionSpecFromDeltaSnapshot(Schema schema) {
+    List<String> partitionNames = deltaLog.snapshot().getMetadata().getPartitionColumns();
+    if (partitionNames.isEmpty()) {
+      return PartitionSpec.unpartitioned();
+    }
+
+    PartitionSpec.Builder builder = PartitionSpec.builderFor(schema);
+    for (String partitionName : partitionNames) {
+      builder.identity(partitionName);
+    }
+    return builder.build();
+  }
+
+  private void copyFromDeltaLakeToIceberg(Table table, PartitionSpec spec) {
+    // TODO: double check the arguments' meaning here
+    Iterator<VersionLog> it =
+        deltaLog.getChanges(
+            0, // retrieve actions starting from the initial version
+            false); // not throw exception when data loss detected
+
+    while (it.hasNext()) {
+      VersionLog versionLog = it.next();
```

Comment:
there is a version number (long) for Delta version log: https://github.com/delta-io/connectors/blob/master/standalone/src/main/java/io/delta/standalone/VersionLog.java#L33

So we should map that as snapshot reference name like `delta-version-<xxxxx>` so that users can do time travel by reference if necessary